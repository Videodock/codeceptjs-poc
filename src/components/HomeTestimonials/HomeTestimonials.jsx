import React from 'react';
import { useTranslation } from 'react-i18next';
import { makeStyles } from '@material-ui/styles';
import { Typography } from '@material-ui/core';
import images from '../../theme/images';
import QuotationMarksIcon from '../Icons/QuotationMarksIcon';
import styles from './HomeTestimonials.styles';

const useStyles = makeStyles(styles);

export default function HomeTestimonials () {
  const { t }   = useTranslation('home');
  const classes = useStyles();

  return (
    <section className={classes.root}>
      <Typography variant="h2">
        {t('testimonials.supported_by')}
      </Typography>
      <div className={classes.container}>
        <div className={classes.testimonialItem}>
          <QuotationMarksIcon />
          <div className={classes.testimonialHead}>
            <img src={images.avatarMahler} alt="" />
            <div>
              <Typography variant="h3">
              G. Mahler
              </Typography>
              <Typography>
                {t('testimonials.composer')}
              </Typography>
            </div>
          </div>
          <Typography>
          "{t('testimonials.quoteMahler')}"
          </Typography>
        </div>
        <div className={classes.testimonialItem}>
          <QuotationMarksIcon />
          <div className={classes.testimonialHead}>
            <img src={images.avatarMozart} alt="" />
            <div>
              <Typography variant="h3">
              W. A. Mozart
              </Typography>
              <Typography>
                {t('testimonials.composer')}
              </Typography>
            </div>
          </div>
          <Typography>
          "{t('testimonials.quoteMozart')}"
          </Typography>
        </div>
        <div className={classes.testimonialItem}>
          <QuotationMarksIcon />
          <div className={classes.testimonialHead}>
            <img src={images.avatarBeethoven} alt="" />
            <div>
              <Typography variant="h3">
              L. von Beethoven
              </Typography>
              <Typography>
                {t('testimonials.composer')}
              </Typography>
            </div>
          </div>
          <Typography>
          "{t('testimonials.quoteBeethoven')}"
          </Typography>
        </div>
      </div>
    </section>
  );
}
