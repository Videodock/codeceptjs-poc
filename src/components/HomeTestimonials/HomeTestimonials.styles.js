import { createStyles } from '@material-ui/styles';
import images from '../../theme/images';

export default ({ spacing, typography, palette, shape, breakpoints }) => createStyles({
  root: {
    display       : 'flex',
    alignItems    : 'center',
    justifyContent: 'center',
    flexDirection : 'column',
    boxSizing     : 'border-box',
    paddingTop    : 98,
    paddingBottom : 98,
    background    : `url(${images.supportedBy})`,
    backgroundSize: 'cover',
    '& h2'        : {
      color: palette.common.white,
    },
    [breakpoints.down('md')]: {
      paddingTop   : 47,
      paddingBottom: 47,
    },
  },
  container: {
    display                 : 'flex',
    alignItems              : 'center',
    justifyContent          : 'center',
    flexWrap                : 'wrap',
    [breakpoints.down('md')]: {
      maxWidth: 400,
    },
  },
  testimonialItem: {
    position       : 'relative',
    marginTop      : 35,
    marginRight    : spacing(2),
    boxSizing      : 'border-box',
    padding        : spacing(3),
    height         : 184,
    width          : 393,
    borderRadius   : shape.borderRadius,
    backgroundColor: palette.common.white,
    boxShadow      : '0 4px 5px 0 rgba(0,0,0,0.14), 0 1px 10px 0 rgba(0,0,0,0.12), 0 2px 4px -1px rgba(0,0,0,0.2)',
    '& svg'        : {
      position: 'absolute',
      top     : -20,
      left    : -5,
      width   : 90,
      height  : 30,
    },
    '&:not($testimonialHead) > p': {
      fontSize: typography.pxToRem(13.82),
    },
    '&:last-child': {
      marginRight: 0,
    },
    [breakpoints.only('xs')]: {
      width : 288,
      height: 'auto',
    },
    [breakpoints.down('md')]: {
      marginRight: 0,
    },
  },
  testimonialHead: {
    display       : 'flex',
    alignItems    : 'center',
    justifyContent: 'flex-start',
    marginBottom  : spacing(3),
    '& img'       : {
      height      : 52,
      width       : 52,
      borderRadius: 60,
      marginRight : spacing(2),
    },
    '& h2': {
      margin: 0,
    },
  },
});
