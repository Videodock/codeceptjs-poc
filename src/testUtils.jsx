import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
