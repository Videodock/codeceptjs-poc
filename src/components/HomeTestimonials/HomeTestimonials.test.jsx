import React from 'react';
import { render } from '../../testUtils';
import HomeTestimonials from './HomeTestimonials';

describe('<HomeTestimonials />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<HomeTestimonials />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
