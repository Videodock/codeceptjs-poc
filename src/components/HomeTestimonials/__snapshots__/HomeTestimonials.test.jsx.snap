// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeTestimonials /> renders and matches snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <section
      class="makeStyles-testimonialsContainer-1"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Supported by
      </h2>
      <div
        class="makeStyles-testimonialsBlock-2"
      >
        <div
          class="makeStyles-testimonial-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="Page-1"
              stroke="none"
              stroke-width="1"
            >
              <g
                fill="#FD820B"
                fill-rule="nonzero"
                id="V3.2-/-1440-/-Landing-page"
                transform="translate(-143.000000, -3006.000000)"
              >
                <g
                  id="Group-2-Copy-2"
                  transform="translate(0.000000, 2820.000000)"
                >
                  <g
                    id="Group-15"
                    transform="translate(96.000000, 156.000000)"
                  >
                    <g
                      id="Group-9-Copy-2"
                      transform="translate(19.000000, 0.000000)"
                    >
                      <path
                        d="M43.6195988,59.5428241 L51.3422068,30.1134259 L37.9841821,30.1134259 L28.591821,59.5428241 L43.6195988,59.5428241 Z M65.9525463,59.5428241 L73.6751543,30.1134259 L60.3171296,30.1134259 L50.9247685,59.5428241 L65.9525463,59.5428241 Z"
                        id="“"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
          <div
            class="makeStyles-testimonialHeader-4"
          >
            <img
              alt=""
              src="avatar.png"
            />
            <div>
              <h3
                class="MuiTypography-root MuiTypography-h3"
              >
                Marcel van Tilburg
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Koninklijk Concertgebouworkest
              </p>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            “The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.”
          </p>
        </div>
        <div
          class="makeStyles-testimonial-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="Page-1"
              stroke="none"
              stroke-width="1"
            >
              <g
                fill="#FD820B"
                fill-rule="nonzero"
                id="V3.2-/-1440-/-Landing-page"
                transform="translate(-143.000000, -3006.000000)"
              >
                <g
                  id="Group-2-Copy-2"
                  transform="translate(0.000000, 2820.000000)"
                >
                  <g
                    id="Group-15"
                    transform="translate(96.000000, 156.000000)"
                  >
                    <g
                      id="Group-9-Copy-2"
                      transform="translate(19.000000, 0.000000)"
                    >
                      <path
                        d="M43.6195988,59.5428241 L51.3422068,30.1134259 L37.9841821,30.1134259 L28.591821,59.5428241 L43.6195988,59.5428241 Z M65.9525463,59.5428241 L73.6751543,30.1134259 L60.3171296,30.1134259 L50.9247685,59.5428241 L65.9525463,59.5428241 Z"
                        id="“"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
          <div
            class="makeStyles-testimonialHeader-4"
          >
            <img
              alt=""
              src="avatar.png"
            />
            <div>
              <h3
                class="MuiTypography-root MuiTypography-h3"
              >
                Marcel van Tilburg
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Koninklijk Concertgebouworkest
              </p>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            “The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.”
          </p>
        </div>
        <div
          class="makeStyles-testimonial-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="Page-1"
              stroke="none"
              stroke-width="1"
            >
              <g
                fill="#FD820B"
                fill-rule="nonzero"
                id="V3.2-/-1440-/-Landing-page"
                transform="translate(-143.000000, -3006.000000)"
              >
                <g
                  id="Group-2-Copy-2"
                  transform="translate(0.000000, 2820.000000)"
                >
                  <g
                    id="Group-15"
                    transform="translate(96.000000, 156.000000)"
                  >
                    <g
                      id="Group-9-Copy-2"
                      transform="translate(19.000000, 0.000000)"
                    >
                      <path
                        d="M43.6195988,59.5428241 L51.3422068,30.1134259 L37.9841821,30.1134259 L28.591821,59.5428241 L43.6195988,59.5428241 Z M65.9525463,59.5428241 L73.6751543,30.1134259 L60.3171296,30.1134259 L50.9247685,59.5428241 L65.9525463,59.5428241 Z"
                        id="“"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </svg>
          <div
            class="makeStyles-testimonialHeader-4"
          >
            <img
              alt=""
              src="avatar.png"
            />
            <div>
              <h3
                class="MuiTypography-root MuiTypography-h3"
              >
                Marcel van Tilburg
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Koninklijk Concertgebouworkest
              </p>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            “The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog. The quick brown fox jumps over the lazy dog.”
          </p>
        </div>
      </div>
    </section>
  </div>
</div>
`;
